### PR TITLE
Update drips, ripples, bubbles

### PR DESCRIPTION
**Pull request by Sezzary**
Rewrite of drip and ripple effect handling:
- Split underwater blood into its own effect (was previously a ripple effect flag).
- Simplified ripple code.
- Simplified drip code.
- Revised player wetness drip handling.
- Improved bubble performance.

New features:
- Player wetness drips create ripples both on water and on land.
- Ripples on land align with the floor's normal.
- Made dive bubbles spawn at player joint nodes, providing more epic theatrics.

NOTE:
- Ripples on the ground don't align with the floor's tilt along a certain axis. This is probably a renderer issue.

### CODE REVIEW

File: TombEngine/Scripting/Internal/TEN/Objects/Lara/LaraObject.cpp [hunk 1]
```diff
@@ -85,8 +85,8 @@ void LaraObject::SetWet(sol::optional<int> wetness)
 {
 	auto* lara = GetLaraInfo(m_item);
 
-	unsigned char value = wetness.has_value() ? (unsigned char)wetness.value() : UCHAR_MAX;
-	for (unsigned char& i : lara->Wet)
+	float value = wetness.has_value() ? (float)wetness.value() : PLAYER_DRIP_NODE_MAX;
```
Comment (Kubsy) [inline TombEngine/Scripting/Internal/TEN/Objects/Lara/LaraObject.cpp:88]: shall I test this?

File: TombEngine/Game/effects/drip.cpp [hunk 2]
```diff
@@ -1,130 +1,173 @@
...
-			auto dripPos = pos + Vector3(Random::GenerateFloat(-128, 128), Random::GenerateFloat(-128, 128), Random::GenerateFloat(-128, 128));
-			auto direction = (dripPos - pos);
-			direction.Normalize();
+			if (drip.Life <= 0.0f)
+				continue;
+
+			// Update velocity.
+			drip.Velocity.y += drip.Gravity;
+			if (TestEnvironment(ENV_FLAG_WIND, drip.RoomNumber))
+				drip.Velocity += Weather.Wind();
```
Comment (Kubsy) [inline TombEngine/Game/effects/drip.cpp:111]: this line caused crash for Adngel

Comment (Sezzary) [reply TombEngine/Game/effects/drip.cpp:111]: Odd, a similar block was used in the old drip implementation. Will look into it.

File: TombEngine/Scripting/Internal/TEN/Objects/Lara/LaraObject.cpp (hunk 1 above)
Comment (Sezzary) [reply TombEngine/Scripting/Internal/TEN/Objects/Lara/LaraObject.cpp:88]: Yes please. The Lua function should also be modified to work with floats.

**Comment by Adngel** [comment]
Need to finishe the review, but right now I've experiencing two important issues.

- Crash when Lara jumps into a water pool.
The debugger tells me the crash is originated in code Drip.cpp line 110. 

```c++
if (TestEnvironment(ENV_FLAG_WIND, drip.RoomNumber))
	drip.Velocity += Weather.Wind();
```

If I comment that, then the crash happens in the next line

```c++
auto pointColl = GetCollision(drip.Position.x, drip.Position.y, drip.Position.z, drip.RoomNumber);```

If I put a return; at beggining of UpdateDrips() function, crashes do not happen anymore. 


- Slow FPS when Lara jumps into a deep water pool. (3 clicks or deeper).
I think it may be related to the bubbles spawn. However I got surprised becase my desktop computer is powerful is very unusual I've get such FPS falls. It last a few, when the bubbles dissapear, the speed returns to normal.

**Comment by Sezzary** [comment]
Might have to wait on Monty's renderer branch (which improves sprite instancing) because of the lag some people are experiencing. Worst case, the bubble count and other things can be adjusted to save FPS.

**Pull request merged**